packages/typescript-estree/src/create-program/getParsedConfigFile.ts

```diff
@@ -0,0 +1,65 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import type * as ts from 'typescript/lib/tsserverlibrary';
+
+import { CORE_COMPILER_OPTIONS } from './shared';
+
+/**
+ * Utility offered by parser to help consumers parse a config file.
+ *
+ * @param configFile the path to the tsconfig.json file, relative to `projectDirectory`
+ * @param projectDirectory the project directory to use as the CWD, defaults to `process.cwd()`
+ */
+function getParsedConfigFile(
+  tsserver: typeof ts,
+  configFile: string,
+  projectDirectory?: string,
+): ts.ParsedCommandLine | string {
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (tsserver.sys === undefined) {
+    throw new Error(
+      '`getParsedConfigFile` is only supported in a Node-like environment.',
+    );
+  }
+
+  const getCurrentDirectory = (): string =>
+    (projectDirectory && path.resolve(projectDirectory)) || process.cwd();
+
+  let parsingError: string | undefined;
+
+  const parsed = tsserver.getParsedCommandLineOfConfigFile(
+    configFile,
+    CORE_COMPILER_OPTIONS,
+    {
+      onUnRecoverableConfigFileDiagnostic: diag => {
+        parsingError = formatDiagnostics([diag]); // ensures that `parsed` is defined.
+      },
+      fileExists: fs.existsSync,
+      getCurrentDirectory,
+      readDirectory: tsserver.sys.readDirectory,
+      readFile: file => fs.readFileSync(file, 'utf-8'),
+      useCaseSensitiveFileNames: tsserver.sys.useCaseSensitiveFileNames,
+    },
+  );
+
+  if (parsed?.errors.length) {
+    parsingError = formatDiagnostics(parsed.errors);
```

Comment:
Question - is it possible for TS to emit an error from the unrecoverable diagnostics hook AND return errors via `parsed.errors`

If yes - then it's possible for us to clobber and lose those unrecoverable diagnostics thanks to this reassignment.

Suggestion: use two separate variables to remove the possibility.

Reply:
If we throw (below comment) it would just be the first that occurred.  Depending on the response to [my question](https://github.com/typescript-eslint/typescript-eslint/pull/9306#discussion_r1690295031) in that thread, it will dictate the direction here.

Reply:
Reverted to throwing.